pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -218,6 +218,12 @@ type InfinispanExternalDependencies struct {
 	// Name of the persistent volume claim with custom libraries
 	// +optional
 	VolumeClaimName string `json:"volumeClaimName,omitempty"`
+	// +optional
+	ExternalLibraries []InfinispanExternalDependencyLibrary `json:"externalLibraries,omitempty"`
```

Comment:
Files placed in `server/libs` don't have to be libraries. WDYT about keeping this simple and just having:

```
Remote []string `json:"remote, omitempty"`
```

This means the user would just do:

```yaml
spec:
  dependencies:
    remote:
      - http://example.com/lib1.jar
      - http://example.com/lib2.jar
```

Reply:
Agree with naming like `remote` or `artifacts` (Strimzi using this type name)
Yeah, original implementation was simple like you provided, but if we will plan to have additional entries, like explicitly defined type (i.e. `zip`, `tgz`, etc.), sha512sum, etc it will be not possible to extend without migration.

Please see how AMQ Streams (Strimzi) defines plugin downloads
https://access.redhat.com/documentation/en-us/red_hat_amq/2021.q2/html-single/using_amq_streams_on_openshift/index#type-Build-reference
```
artifacts:
  - type: tgz
  url: https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/1.3.1.Final/debezium-connector-postgres-1.3.1.Final-plugin.tar.gz
  sha512sum: 962a12151bdf9a5a30627eebac739955a4fd95a08d373b86bdcea2b4d0c27dd6e1edd5cb548045e115e33a9e69b1b2a352bee24df035a0447cb820077af00c03
```

Reply:
Thanks for the Strimzi example. I didn't consider optional sha's before ... that would be a great addition.

Do we really need sha-512 for verification? I think sha-256 is what's used in most places, e.g. container images.

How about we make it so the `type` and `sha` fields are optional. If no type is specified, then we can use the extension to determine how to handle the file, however if the type is provided we always process the artifact in that manner. Initially we could provide support for `type = file | zip | tgz` as you have already implemented this.

For example:

```yaml
spec:
  dependencies:
    artifacts:
      # Copy file as is
      - url: http://example.com/lib1.jar
      # Extract jar contents
      - url: http://example.com/lib1.jar
        type: zip
      # Don't extract as type explicitly configured
      - url: http://example.com/lib1.zip
        type: file
      # Extract as no type, but .zip extension
      - url: http://example.com/lib1.zip
```

Reply:
> Do we really need sha-512 for verification? I think sha-256 is what's used in most places, e.g. container images.

I'm not sure, but probably `sha-1` is enough for validation in most cases, I see it uses for distribution download validation on infinispan.org as well as for maven also in addition to the `md5` @ryanemerson wdyt about it?

Reply:
I don't have strong views on what sha mechanism should be used for verification (any thoughts @tristantarrant @Crumby), but I think we should have this available before merging as it could save a lot of trouble when debugging user cases.

Reply:
I think `sha1` is sufficient.
A couple of notes:
* Should we only allow `https` by default ? Maybe with a `allowInsecureUrls: true` option to override
* Do we support credentials ?
* It would be awesome if we also supported Maven GAVs:
` - maven: groupId:artifactId:version[:packaging[:classifier]]]`
This should default to Maven central https://repo1.maven.org/maven2/ with the possibility of overriding the repo

Reply:
> * This should default to Maven central https://repo1.maven.org/maven2/ with the possibility of overriding the repo

IMO per Infinispan CR overriding would be overkill, but we could allow a maven "settings.xml" to be provided as part of the global operator configuration.

Reply:
We can probably add real maven support to the OCP Build (k8s Kaniko build) option and build a real image just one time
Also it's possible to download maven artifacts with the direct URL link. Credential can be also provided via URL (not secure way), but probably will be always insecure with Init containers utilization. The single secure way us using builds

Reply:
I've added any hash algorithms to be available for hashing (sha1|sha224|sha256|sha384|sha512|md5)